core/src/replay_stage.rs

```diff
@@ -3867,8 +3871,42 @@ impl ReplayStage {
                         fork_stake,
                         lockout_intervals,
                         my_latest_landed_vote,
+                        parent_is_super_oc,
                         ..
                     } = computed_bank_state;
+
+                    if parent_is_super_oc
+                        && migration_status.qualifies_for_genesis_discovery(bank_slot)
+                    {
+                        let migration_slot = migration_status.migration_slot();
+                        // We have a block whose ancestor qualifies to be the genesis block.
+                        let genesis_slot = ancestors
+                            .get(&bank_slot)
+                            .expect(
+                                "Ancestors must exist, as this cannot be slot 0, and no roots are \
+                                 made during migration",
+                            )
+                            .iter()
+                            .filter(|slot| **slot < migration_slot)
+                            .max()
+                            .copied()
+                            .expect("Genesis slot must exist, no rooting past migration slot");
+                        let genesis_bank = bank_forks
+                            .read()
+                            .unwrap()
+                            .get(genesis_slot)
+                            .expect("Genesis bank must exist, no rooting past migration slot");
+
+                        let genesis_block = (
+                            genesis_slot,
+                            genesis_bank.block_id().expect(
+                                "It is impossible for block id to not be known at this point, as \
+                                 a descendant of this block has reached super oc status",
+                            ),
+                        );
+                        migration_status.set_genesis_block(genesis_block);
```

Comment:
slightly off topic, but I believe we used to have an assert in this function to check if the genesis block (slot number) was in expected range. Might be nice to have that condition somewhere to catch any oversights when we start hammering this in testing